beta-src/src/utils/map/drawBuilds.ts

```diff
@@ -30,12 +31,10 @@ export default function drawBuilds(state): void {
     maps: GameState["maps"];
     ownUnits;
   } = current(state);
+
   if (phase === "Builds") {
     ownUnits.forEach((unitID) => {
-      const command: GameCommand = {
-        command: "NONE",
-      };
-      setCommand(state, command, "unitCommands", unitID);
+      state.unitState[unitID] = UIState.NONE;
```

Comment:
Would it be possible to have a class so that we could abstract this? I think it would be a good investment to control the way we are allowed to update this state. The commands thing did this but we can take an approach without it. So maybe we'd have a class called `UpdateUnit` (or  a better name) and this class could have a method that takes a specific type to update the unit Icon state. 

Dirty example:
```
class UpdateUnit extends UpdateGame {
   setUIState(newIcon: UIState){
       state.unitState[unitID] = UIState.NONE;
   }
}
```

Reply:
Potential usage: 

```
const updater = new UpdateUnit(state, unitID);
updater.setUIState(UIState.HOLD);
```

Reply:
Alternatively, maybe an object of functions could work using object composition? If we prefer not to go with classes.

Reply:
Yeah, that makes sense. I'm not yet sure if `unitState` is the representation we want long-term so I don't want to spend more time building abstractions around it, but if I stick with it then I'll do that.